### ISSUE TITLE
[feat] Create `<rh-autocomplete>` component/pattern

### ISSUE THREAD

**Issue by kelsS** [RedHat-UX/red-hat-design-system#1986]
### Description

Hi there, we need an RHDS version of [pfe-autocomplete](https://patternflyelements.org/v1/components/autocomplete/) that passes a11y tests. The current `pfe-autocomplete` has contrast issues that cannot be fixed since v1 is deprecated but there is no other pfe version or RHDS version of autocomplete that can be used instead yet.

This is an important component that I think should be in RHDS. 

### Acceptance Criteria

- [ ] Design done
- [ ] Development done

### Other resources

- https://www.w3.org/WAI/ARIA/apg/patterns/combobox/
- https://patternflyelements.org/v1/components/autocomplete/

**Comment by kelsS** [comment]
@hellogreg I wanted you to see this issue as it pertains to the [Jira ticket](https://issues.redhat.com/browse/CPCORE-11852) you created for the CP Home page.

**Comment by hellogreg** [comment]
Closed in favor of #367
